### DISCUSSION TITLE
Uncaught TypeError: Cannot read properties of null (reading 'useContext') when using AntDesign

### DISCUSSION THREAD

**Discussion by SergeySh108** [ant-design/ant-design#39720] [Q&A]
I was implement UI design library (flatty-ui) based on AntDesign with react app
Example element from library
import React, {FC, ReactElement} from 'react';
import { Button } from "antd";
import "./Button.scss"

const FlattyButton: FC<ButtonProps> = () => {
    return (
        <Button> Button </Button>
    );
};

export default FlattyButton;

Other project based on react app too. I import this library (for example: import { FlattyButton } from "flatty-ui") and then use it in code <FlattyButton ...>...

import React from 'react';
import { BrowserRouter } from "react-router-dom";
import {
  QueryClient,
  QueryClientProvider,
  useQuery,
} from 'react-query';
import {FlattyButton} from 'flatty-ui';
import './App.css';

const queryClient = new QueryClient();

function App() {
  return (
    <>
        <QueryClientProvider client={queryClient}>
       <BrowserRouter>
           <FlattyButton></FlattyButton>
       </BrowserRouter>
       </QueryClientProvider>
    </>
  );
}

export default App;

I run my code  with npm run start and have error 
-------------
react.development.js:209 Warning: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
printWarning @ react.development.js:209
react.development.js:1618 Uncaught TypeError: Cannot read properties of null (reading 'useContext')

Uncaught TypeError: Cannot read properties of null (reading 'useContext')
-------------

Could somebody explain what is it?

**Comment by SignDawn** [comment]
```
const FlattyButton: FC = () => {
return (
<Button />
);
};
```

**Comment by SignDawn** [comment]
React 元素于 React 函数的概念要区分下
